projects/uswds-components/src/lib/textarea/textarea.component.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  Optional,
+  Output,
+  Self,
+} from '@angular/core';
+import { ControlValueAccessor, NgControl } from '@angular/forms';
+import { FormFieldComponent } from '../form-field/form-field.component';
+import { Key } from '../util/key';
+
+let nextId = 0;
+
+@Component({
+  selector: 'usa-textarea',
+  templateUrl: './textarea.component.html',
+})
+export class UsaTextareaComponent implements ControlValueAccessor {
+  model: string = '';
+
+  private _onChange = (_: any) => {};
+  private _onTouched = () => {};
+  /**
+   * Sets the ariaLabel attribute
+   */
+  @Input() ariaLabel = 'Textarea component';
+  /**
+   * Sets the rowHeight
+   */
+  @Input() rowHeight: number;
+  /**
+   * Sets the maxLength attribute
+   */
+  @Input() maxlength: number;
+
+  /**
+   * Sets the maxLength attribute
+   */
+  @Input() minlength: number;
+
+  /**
+   * Sets the id attribute
+   */
+  @Input() id = `usa-textarea-${nextId++}`;
+  /**
+   * Sets the placeholder attribute
+   */
+  @Input() placeholder = '';
+  /**
+   * Sets the name attribute
+   */
+  @Input() name = 'textarea';
+
+  /**
+   * Optional assistance text to be set when placeholder attribute is used
+   */
+  @Input() title: string;
+
+  /**
+   * Sets the disabled attribute
+   */
+  @Input() disabled: boolean;
+
+  /**
+   * Emits focus event
+   */
+  @Output() onBlur: EventEmitter<string> = new EventEmitter(null);
+
+  /**
+   * deprecated, emits value change events
+   */
+  @Output() valueChange: EventEmitter<string> = new EventEmitter<string>();
+
+  /**
+  * Sets the required attribute
+
+  */
+  @Input() required: boolean; // deprecated
+  /**
+   * Sets the required attribute
+   */
+  @Input() requiredFlag: boolean;
+
+  /**
+   * Sets the label text
+   */
+  @Input() label: string;
+
+  /**
+   * Sets the helpful description text
+   */
+  @Input() description: string;
+  /**
+   * Sets the general error message
+   */
+  @Input() errorMessage: string;
+
+  @Input() hasWrapper = false;
```

Comment:
is this hasWrapper boolean useful? It seems like we're doing a check on whether wrapperRef is undefined or not to achieve the same check

Reply:
forgot to remove it